### PR TITLE
Feature/new datalake endpoints

### PR DESCRIPTION
**Pull request by ignacioFernandez1**

### CODE REVIEW

File: splight_lib/client/datalake/remote_client.py [hunk 1]
```diff
@@ -149,61 +172,6 @@ def delete(self, collection: str, **kwargs) -> None:
...
-            dataframe.to_csv(fid)
-        response = self._restclient.post(
-            url,
-            data={"source": collection},
-            files={"file": open(tmp_file.name, mode="rb")},
-        )
-        response.raise_for_status()
-
     @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
     def create_index(self, collection: str, indexes: List[Dict]) -> None:
```
Comment (matisilva) [inline splight_lib/client/datalake/remote_client.py:176]: Remove this function

File: splight_lib/client/datalake/abstract.py [hunk 2]
```diff
@@ -54,11 +54,11 @@ def save_dataframe(self, collection: str, dataframe: pd.DataFrame) -> None:
...
-    @abstractmethod
-    def raw_aggregate(
-        self, collection: str, pipeline: List[Dict]
-    ) -> List[Dict]:
-        pass
+    # @abstractmethod
+    # def raw_aggregate(
+    #     self, collection: str, pipeline: List[Dict]
+    # ) -> List[Dict]:
+    #     pass
```
Comment (matisilva) [inline splight_lib/client/datalake/abstract.py:61]: remove this

File: splight_lib/client/datalake/local_client.py [hunk 3]
```diff
@@ -138,12 +138,12 @@ def create_index(self, collection: str, index: list) -> None:
...
-    ) -> List[Dict]:
-        logger.debug(
-            "Skipping raw aggregation when using Local datalake client."
-        )
+    # def raw_aggregate(
+    #     self, collection: str, pipeline: List[Dict]
+    # ) -> List[Dict]:
+    #     logger.debug(
+    #         "Skipping raw aggregation when using Local datalake client."
+    #     )
```
Comment (matisilva) [inline splight_lib/client/datalake/local_client.py:146]: Remove

File: splight_lib/client/datalake/remote_client.py [hunk 4]
```diff
@@ -62,84 +65,104 @@ async def async_save(
...
+        response = self._restclient.post(url, json=data_request.dict())
         response.raise_for_status()
-        output = response.json()["results"]
+        output = response.json()
 
         return output
 
     @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    async def _async_raw_get(
+    async def _async_get(
```
Comment (matisilva) [inline splight_lib/client/datalake/remote_client.py:129]: Why we need a sync and async method? IMO always is better to have one interface to fetch one to write (agnostic of implementation)

**Review by matisilva — CHANGES_REQUESTED** [review]

File: splight_lib/client/datalake/remote_client.py [hunk 5]
```diff
@@ -61,217 +67,116 @@ async def async_save(
...
-        to_timestamp: Optional[datetime],
-        query: Dict,
-    ) -> pd.DataFrame:
-        url = self._base_url / f"{self._PREFIX}/data/request/csv/"
-        to_timestamp = to_timestamp.isoformat() if to_timestamp else None
-        data = {
-            "from_timestamp": from_timestamp.isoformat(),
-            "to_timestamp": to_timestamp,
-            "traces": [
+        # GET /data/read
```
Comment (francozacco) [inline splight_lib/client/datalake/remote_client.py:122]:
```suggestion
        # POST /data/read
```

**Review by francozacco — APPROVED** [review]

File: splight_lib/client/datalake/remote_client.py (hunk 5 above)
Comment (marianogaragiola) [reply splight_lib/client/datalake/remote_client.py:122]: No need for this, the endpoint is a POST

Comment (francozacco) [reply splight_lib/client/datalake/remote_client.py:122]: yeah I know, I just wanted to say that the comment was wrong

**Review by smartinellimarco — APPROVED** [review]

**Review by marianogaragiola — APPROVED** [review]

**Review by matisilva — APPROVED** [review]

**Pull request merged**